lib/elements/fill_stitch.py

```diff
@@ -699,10 +724,25 @@ def to_stitch_groups(self, previous_stitch_group):  # noqa: C901
             return self.do_legacy_fill()
         else:
             stitch_groups = []
-            end = self.get_ending_point()
 
-            for shape in self.shape.geoms:
+            # start and end points
+            start = self.get_starting_point(previous_stitch_group)
+            final_end = self.get_ending_point()
+
+            # sort shapes to get a nicer routing
+            shapes = list(self.shape.geoms)
+            if start:
+                shapes.sort(key=lambda shape: shape.distance(shgeo.Point(start)))
+            else:
+                shapes.sort(key=lambda shape: shape.bounds[0])
```

Comment:
No, but I'm not sure what else to use. So this at least is some sort of reordering which works for a lot but definitely not always. So I thought we could go for the x-axis only for now and add a better routing later on in an other branch?

Reply:
Sure, that's fine with me.  If your goal is to sort from left to right along the direction of the fill stitch lines, I think we can do that by rotating each shape by `-angle` before getting their bounding boxes.  I'm not sure though.

Reply:
Hm, using the stitch direction seems to be an interesting thought. It could still lead to back and forth jumping between the connected shapes. They could be on top of each other either way (in stitch direction or normal bounding box). So I'm not sure if it is worth the calculation time.